distributed/client.py

```diff
@@ -2992,7 +2994,7 @@ def run(
     @staticmethod
     def _get_computation_code(
         stacklevel: int | None = None, nframes: int = 1
-    ) -> tuple[str, ...]:
+    ) -> tuple[SourceCode, ...]:
```

Comment:
Assuming we want to use this data, whatever type we return here is going to be what gets passed to `update_graph` 

https://github.com/dask/distributed/blob/fbab38adc1f3174f64347102492f3302aa8eddae/distributed/scheduler.py#L4269

and used as `Computation.code`, which then needs to be read/parsed/used by the analytics plugin for this to be useful on the Coiled platform side.

Given that it's a pain to coordinate schema changes and that tuples don't give a lot of flexibility, what do you think of switching here to a `dict`? We're making a breaking change (from `str` to something else) anyway, so why not make it to something more flexible so that in the future we can avoid breaking changes. From my perspective this would make things easier to maintain.

Reply:
Is it not viable to import the `SourceCode` obj on the platform side? AFAIK, a schema change in the dict would be just as inconvenient as a change in a namedtuple. 

Aside from _really_ not liking to pass around generic dicts, it would also require a change in  `Computation.code` from being a `SotrtedSet` to something else since `dict` isn't hashable.

Reply:
Hm, I guess the plugin could import `SourceCode` to see what's available. Things are more complicated because plugin has to be fairly agnostic about dask/distributed version but this should work (with a fallback to current way of interpreting `code` when `SourceCode` isn't importable).